perceval/backends/core/gitter.py

```diff
@@ -0,0 +1,325 @@
+
+    The origin of the data will be set to the `GITTER_URL` plus the
+    identifier of the room; i.e 'https://gitter.im/{group}/{room}'.
+
+    :param group: group to which the room belongs
+    :param room: identifier of the room from which the messages are to be fetched
+    :param api_token: token or key needed to use the API
+    :param max_items: maximum number of message requested on the same query
+    :param tag: label used to mark the data
+    :param archive: archive to store/retrieve items
+    :param ssl_verify: enable/disable SSL verification
+    """
+    version = '0.1.0'
+
+    CATEGORIES = [CATEGORY_MESSAGE]
+
+    def __init__(self, group=None, room=None, api_token=None, max_items=MAX_ITEMS,
+                 tag=None, archive=None, ssl_verify=True):
+        origin = urijoin(GITTER_URL, group, room)
+
+        super().__init__(origin, tag=tag, archive=archive, ssl_verify=ssl_verify)
+        self.group = group
+        self.room = room
+        self.api_token = api_token
+        self.max_items = max_items
+        self.client = None
+        self.room_id = None
+
+    def search_fields(self, item):
+        """Add search fields to an item.
+
+        It adds the values of `metadata_id`,`group`,`room`
+        and 'room_id'.
+
+        :param item: the item to extract the search fields values
+
+        :returns: a dict of search fields
+        """
+        search_fields = {
+            DEFAULT_SEARCH_FIELD: self.metadata_id(item),
+            'group': self.group,
+            'room': self.room,
+            'room_id': self.room_id
+
+        }
+
+        return search_fields
+
+    def fetch(self, category=CATEGORY_MESSAGE, from_date=DEFAULT_DATETIME):
+        """Fetch the messages from the room.
+
+        This method fetches the messages sent in the room that were
+        sent since the given date.
+
+        :param category: the category of items to fetch
+        :param from_date: date from which messages are to be fetched
+
+        :returns: a generator of messages
+        """
+        if not from_date:
+            from_date = DEFAULT_DATETIME
+
+        from_date = from_date.strftime('%Y-%m-%d')
+        kwargs = {
+            'from_date': from_date,
+        }
+
+        items = super().fetch(category, **kwargs)
+
+        return items
+
+    def fetch_items(self, category, **kwargs):
+        """Fetch the messages
+
+        :param category: the category of items to fetch
+        :param kwargs: backend arguments
+
+        :returns: a generator of items
+        """
+        from_date = kwargs['from_date']
+        logger.debug("Get Gitter message paginated items of room: %s from date: %s", self.room, from_date)
+        from_date = str_to_datetime(from_date).timestamp()
+        fetching = True
+        before_id = None
+        num_msgs = 0
+        self.room_id = self.client.get_room_id()
+        logger.debug("Room Id of room: %s is %s", self.room, self.room_id)
+        page = 0
```

Comment:
Please move the variable `page` before the call of `logger.debug`, and separate them with a blank line to improve readibility.